internal/cli/cmd/cluster/config.go

```diff
@@ -227,8 +227,8 @@ func (r *reconfigureOptions) syncClusterComponent() error {
 	if err != nil {
 		return makeClusterNotExistErr(r.clusterName)
 	}
-	if len(componentNames) != 1 {
-		return cfgcore.MakeError("please specify a component as there are more than one component in cluster.")
+	if len(componentNames) == 0 {
```

Comment:
`componentNames` are all components of current cluster, if the number of component greater than 1, cli DOES NOT know which component to deal with, user need specify one, so I think the original `len(componentNames) != 1` is right.

Reply:
This is a compromise, config observer related subcommand (desc-config/explain-config), is to allow users to quickly see the components and configSpec informations, to help configure the subcommand to set reasonable parameters, may print out the full summary of the information is more friendly to new users!

The default behavior is to print all summary instead of reporting an error, and it is possible that the new user may not know which components are available.

Your feedback is correct, this pr is closed first, and the complete function pr can be referred to here： #4535